### PR TITLE
III-6079 moviefetcher image

### PR DESCRIPTION
**Pull request by JonasVHG**
### Changed

- `KinepolisClient`: Added  `getImage()` to fetch the Image from a movie
- `ParsedMovie`: Added `$poster` as property
- `KinepolisService`: Upload Poster as Image to `UiTdatabank`, when creating a new movie.

### Fixed

- Movies now have an Image.

---
Ticket: https://jira.publiq.be/browse/III-6079

### CODE REVIEW

File: src/Kinepolis/AuthenticatedKinepolisClient.php [hunk 1]
```diff
@@ -72,6 +73,33 @@ public function getMovieDetail(string $token, int $mid): array
         return $contents['movies'][0];
     }
 
+    public function getImage(string $token, string $path): UploadedFile
+    {
+        $lastSlashPosition = strrpos($path, '/');
```
Comment (grubolsch) [inline src/Kinepolis/AuthenticatedKinepolisClient.php:78]: Variable can move inline

File: src/Kinepolis/AuthenticatedKinepolisClient.php [hunk 2]
```diff
@@ -72,6 +73,33 @@ public function getMovieDetail(string $token, int $mid): array
...
+
+        $fileName = substr($path, $lastSlashPosition + 1);
```
Comment (grubolsch) [inline src/Kinepolis/AuthenticatedKinepolisClient.php:80]: I would move this a separate function with a function name that explains what / why you are doing this.

File: src/Kinepolis/AuthenticatedKinepolisClient.php [hunk 3]
```diff
@@ -72,6 +73,33 @@ public function getMovieDetail(string $token, int $mid): array
...
+        $headers['Accept'] = 'image/jpeg';
+        $request = new Request(
+            'GET',
+            $this->movieApiBaseUrl . $path,
+            $headers
+        );
+
+        $response = $this->client->sendRequest($request);
+        $imageStream = $response->getBody();
+        $mimeType = $response->getHeaderLine('Content-Type');
```
Comment (grubolsch) [inline src/Kinepolis/AuthenticatedKinepolisClient.php:92]: Can be moved inline

**Review by grubolsch — APPROVED** [review]

File: src/Kinepolis/ParsedMovie.php [hunk 4]
```diff
@@ -27,14 +27,17 @@ final class ParsedMovie
...
     public function __construct(
         string $externalId,
         Title $title,
         LocationId $locationId,
         Description $description,
         Theme $theme,
         Calendar $calendar,
-        PriceInfo $priceInfo
+        PriceInfo $priceInfo,
+        string $poster
```
Comment (LucWollants) [inline src/Kinepolis/ParsedMovie.php:40]: What is poster? Is that the URL of the image?

Comment (JonasVHG) [reply src/Kinepolis/ParsedMovie.php:40]: Indeed,  I kept the naming consistent with the naming in the JSON API from Kinepolis.

File: src/Kinepolis/AuthenticatedKinepolisClient.php [hunk 5]
```diff
@@ -72,6 +73,33 @@ public function getMovieDetail(string $token, int $mid): array
...
+
+        return new UploadedFile(
```
Comment (LucWollants) [inline src/Kinepolis/AuthenticatedKinepolisClient.php:94]: I'm a bit confused by the usage of the `UploadedFile` class here. Normally an `UploadedFile` is something that was uploaded to the UDB3 server and is physically present on the file system. In this case, it is a stream from another online URL, will this work?

Comment (JonasVHG) [reply src/Kinepolis/AuthenticatedKinepolisClient.php:94]: I tested it locally, and it worked, I will have to test it again, when it is on the server.

File: src/Kinepolis/KinepolisService.php [hunk 6]
```diff
@@ -173,6 +182,19 @@ private function createNewMovie(ParsedMovie $parsedMovie): string
...
+        // to avoid multiple needless updates, we only add an image during the creation
+        $uploadedImage = $this->client->getImage($token, $parsedMovie->getPoster());
+        $imageId = $this->imageUploader->upload(
+            $uploadedImage,
+            new MediaDescription($parsedMovie->getTitle()->toString()),
+            new CopyrightHolder('Kinepolis'),
+            new LegacyLanguage('nl')
+        );
+        $addImage = new AddImage($eventId, $imageId);
+        $this->commandBus->dispatch($addImage);
```
Comment (LucWollants) [inline src/Kinepolis/KinepolisService.php:196]: It's a bit weird that the `createNewMovie` also does the dispatch of the `AddImage` command. Maybe that needs to become a separate method that can be called inside the `process` method. That new method can do the upload of the file and returns the `AddImage` command.

File: src/Kinepolis/ParsedMovie.php (hunk 4 above)
Comment (LucWollants) [reply src/Kinepolis/ParsedMovie.php:40]: I have no strong opinion about it, but is image more clear in the UDB3 context?

File: src/Kinepolis/AuthenticatedKinepolisClient.php (hunk 5 above)
Comment (LucWollants) [reply src/Kinepolis/AuthenticatedKinepolisClient.php:94]: Then it will be fine because it is a stream object.

**Review by LucWollants — APPROVED** [review]

**Pull request merged**